### PR TITLE
avoid calling initModule twice on service start, and make sure the mo…

### PR DESCRIPTION
**Pull request by wei-lee**
…ngodb connection is closed when an export job is completed

### CODE REVIEW

File: fh-mbaas.js [hunk 1]
```diff
@@ -204,7 +206,11 @@ function getMbaasMiddlewareConfig() {
 }
 
 function initModules(clusterWorker, jsonConfig, cb) {
-
+  if(initModulesCalled) {
+    return process.nextTick(() => cb());
+  } else {
```
Comment (sedroche) [inline fh-mbaas.js]: Do you need the else statement?

File: fh-mbaas.js [hunk 2]
```diff
@@ -39,6 +39,8 @@ var mongooseConnection;
 
 var START_AGENDA = "startAgenda";
 
+var initModulesCalled = false;
```
Comment (pb82) [inline fh-mbaas.js]: but it is ok if `initModules` is called once per worker right?

File: fh-mbaas.js [hunk 3]
```diff
@@ -223,6 +229,7 @@ function initModules(clusterWorker, jsonConfig, cb) {
     if (!err) {
       return cb();
     }
+    initModulesCalled = false;
```
Comment (pb82) [inline fh-mbaas.js]: Why set it back to false here? Does that not mean it will be executed every time?

File: fh-mbaas.js (hunk 1 above)
Comment (wei-lee) [inline fh-mbaas.js]: no, we don't...

File: fh-mbaas.js (hunk 2 above)
Comment (wei-lee) [inline fh-mbaas.js]: yes, that's right. Each worker should only need to call it once

File: fh-mbaas.js (hunk 3 above)
Comment (wei-lee) [inline fh-mbaas.js]: if the case of failure, allow this to be called again.

**Comment by wei-lee** [comment]
Retest this please

**Comment by wei-lee** [comment]
Retest this please

**Pull request merged**